src/Simplex/Messaging/Server.hs

```diff
@@ -1854,7 +1870,7 @@ client
                       ProhibitSub -> pure Nothing
                       ServerSub st -> readTVar st >>= \case
                         NoSub ->
-                          tryTakeTMVar delivered >>= \case
+                          tryReadTMVar delivered >>= \case
```

Comment:
This looks like an incorrect logic that had no effect as it _should not be_ possible to have concurrent deliveries to an empty queue (see `wasEmpty` condition above). But if it were somehow possible, it would have resulted in stuck delivery.